src/toolbox.js

```diff
@@ -30,53 +30,92 @@ function onConnect({client} = {}) {
     return;
   }
 
-  Promise.all(exps.map(ex => client.clientCommands.evaluate(ex, {}))).then(results => {
-      ReactDOM.render(
-        dom.div({},
-          results.map(result =>
-            dom.div(
-              {className: "rep-row", key:  JSON.stringify(result)},
-              Object.keys(MODE).map(modeKey =>
-               renderRep({ object: result.result, modeKey })
-             )
-           )
-         )
-      ),
-      root
-    )
-  })
+  ReactDOM.render(
+    (React.createFactory(RepsConsole))({client}),
+    root
+  );
 }
 
-function renderRep({ object, modeKey }) {
-  return dom.div(
-    {
-      className: `rep-element ${modeKey}`,
-      key: JSON.stringify(object) + modeKey
-    },
-    Rep({ object, defaultRep: Grip, mode: MODE[modeKey] })
-  )
-}
+RepsConsole = React.createClass({
+  getInitialState: function() {
+    return {
+      expressions: []
```

Comment:
This is an awesome start. 

I'm tempted to do a second pass where you have some links like "arrays", "times", ... where it runs all of the common examples. then you can add special cases with the input